### PR TITLE
Add `cupy.poly1d.__pow__`

### PR DESCRIPTION
**Pull request by Dahlia-Chehata**
I don't think the loop can be parallelized right?

### CODE REVIEW

File: cupy/lib/polynomial.pyx [hunk 1]
```diff
@@ -161,11 +161,13 @@ cdef class poly1d:
...
         return _routines_poly.polyadd(self, other)
 
-    # TODO(Dahlia-Chehata): implement using polymul
     def __pow__(self, val, modulo):
         if not cupy.isscalar(val) or int(val) != val or val < 0:
             raise ValueError('Power to non-negative integers only.')
-        raise NotImplementedError
+        out = 1
+        for _ in range(val):
+            out = _routines_poly.polymul(self, out)
```
Comment (asi1024) [inline cupy/lib/polynomial.pyx:169]: How about trying O(log(val)) time convolve calls, and comparing performances?

Comment (asi1024) [reply cupy/lib/polynomial.pyx:169]: note: https://en.wikipedia.org/wiki/Exponentiation_by_squaring

Comment (Dahlia-Chehata) [reply cupy/lib/polynomial.pyx]: I'll make performance comparison some time later.

**Comment by Dahlia-Chehata** [comment]
```
val=10
NumPy:  __pow__ :    CPU:  285.606 us   +/-31.281 (min:  267.980 / max:  964.118) us     GPU-0:  249.355 us   +/-57.652 (min:   12.032 / max:  976.000) us
CuPy log:  __pow__ :    CPU:  422.528 us   +/-60.135 (min:  391.981 / max: 1204.119) us     GPU-0:  420.272 us   +/-103.282 (min:  239.296 / max: 2931.904) us
CuPy linear:  pow2 :    CPU: 3791.435 us   +/-450.935 (min: 3171.240 / max:11529.336) us     GPU-0: 3799.698 us   +/-446.314 (min: 3181.888 / max:11548.800) us

val=100
NumPy:  __pow__ :    CPU: 2612.061 us   +/-118.630 (min: 2475.255 / max: 6109.650) us     GPU-0: 2620.900 us   +/-122.529 (min:  650.496 / max: 6119.776) us
CuPy log: __pow__ :    CPU:  786.774 us   +/-49.332 (min:  746.983 / max: 1480.116) us     GPU-0:  794.463 us   +/-55.483 (min:  754.208 / max: 3208.768) us
CuPy linear: pow2 :    CPU:25032.412 us   +/-466.870 (min:24366.896 / max:32271.256) us     GPU-0:25043.023 us   +/-466.932 (min:24377.024 / max:32282.879) us

val=1000
NumPy:  __pow__:    CPU:30651.120 us   +/-530.765 (min:29851.922 / max:55339.203) us     GPU-0:30673.198 us   +/-532.330 (min:29869.921 / max:55377.632) us
CuPy log: __pow__ :    CPU: 1401.218 us   +/-78.952 (min: 1316.168 / max: 2966.183) us     GPU-0: 3603.564 us   +/-86.382 (min: 3494.496 / max: 6388.576) us
CuPy linear: pow2 :    CPU:253086.137 us   +/-19139.485 (min:243690.534 / max:517671.570) us     GPU-0:253098.118 us   +/-19139.822 (min:243703.873 / max:517689.026) us
```
Script
```
def compare(n):
    def generator(xp):
        x = xp.array([random.randint(1, 1000) for _ in range(5)])
        return xp.poly1d(x)
    x1 = generator(np)
    x2 = generator(cp)

    np_ret = repeat(np.poly1d.__pow__, (x1, n))
    cp_ret = repeat(cp.poly1d.__pow__, (x2, n))
    cp_ret2 = repeat(cp.poly1d.pow2, (x2, n))

    print("NumPy: ", np_ret.to_str(show_gpu=True))
    print("CuPy log: ", cp_ret.to_str(show_gpu=True))
    print("CuPy linear: ", cp_ret2.to_str(show_gpu=True))
```

File: cupy/lib/_routines_poly.py [hunk 2]
```diff
@@ -104,6 +104,16 @@ def polymul(a1, a2):
     return cupy.convolve(a1, a2)
 
 
+def _polypow(x, n):
+    if n == 0:
+        return 1
+    if n == 1:
+        return x
+    if n % 2 == 0:
+        return _polypow(cupy.convolve(x, x), n / 2)
```
Comment (asi1024) [inline cupy/lib/_routines_poly.py:113]: Please use integer division.
```suggestion
        return _polypow(cupy.convolve(x, x), n // 2)
```

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**